protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -73,6 +73,35 @@ definitions:
       emitted_at:
         description: "when the data was emitted from the source. epoch in millisecond."
         type: integer
+      meta:
+        description: Information about this record added mid-sync
+        "$ref": "#/definitions/AirbyteRecordMessageMeta"
+  AirbyteRecordMessageMeta:
+    type: object
+    additionalProperties: true
+    properties:
+      errors:
+        description: Key-Value pairs of problems for properties of this record
+        type: array
+        items:
+          "$ref": "#/definitions/AirbyteRecordMessageMetaErrors"
+  AirbyteRecordMessageMetaErrors:
+    type: object
+    additionalProperties: true
+    required:
+      - key
+      - message
+    properties:
+      key:
```

Comment:
I also just realized - platform is _most_ likely to have record-level errors (i.e. "this record is >20MB"). Per-field errors (e.g. redshift's "this string is >64KB") are probably more in the realm of destination connector errors.

Reply:
I think that case is still possible, and can be solved by this message structure.  The algorithm for which fields to keep nulling until the total record size is small enough is up to the implementation (e.g. `destination-redshift`), but then we'd end up with messages like:

```js
    meta: {
      errors: [
        { property: "big-property-1", message: "property `big-property-1` was nulled to ensure total record size was 16mb or less", origin: "destination" }, 
        { property: "big-property-2", message: "property `big-property-2` was nulled to ensure total record size was 16mb or less", origin: "destination" }
      ]
    } 
```

If I had to guess what the algorithm for redshift would be it would be:

```
while total-size > 16mb:
  properties_to_consider = properties.map(p -> p is not cursor && p is not primary_key)
  if (properties_to_consider.length === 0) throw new Error("can't truncate record")
  largest_property = properties_to_consider.sort((a,b) -> a.size > b.size).first
  largest_property.value = null
```

Reply:
makes sense!